src/engraving/libmscore/measure.cpp

```diff
@@ -4258,6 +4239,31 @@ void Measure::computeWidth(Fraction minTicks, qreal stretchCoeff)
     bool isSystemHeader = s->header();
 
     computeWidth(s, x, isSystemHeader, minTicks, stretchCoeff);
+
+    // Check against minimum width and increase if needed
+    double minWidth = isMMRest() ? score()->styleMM(Sid::minMMRestWidth) : score()->styleMM(Sid::minMeasureWidth);
+    double maxWidth = system()->width() - system()->leftMargin(); // maximum available system width (left margin accounts for possible indentation)
+    minWidth = std::min(minWidth, maxWidth); // Accounts for a case where the user may set the minMeasureWidth to a value larger than the available system width
+    if (width() < minWidth) {
+        setWidthToTargetValue(s, x, isSystemHeader, minTicks, stretchCoeff, minWidth);
+        setWidthLocked(true);
+    } else {
+        setWidthLocked(false);
+    }
+}
+
+void Measure::setWidthToTargetValue(Segment* s, qreal x, bool isSystemHeader, Fraction minTicks, qreal stretchCoeff, qreal targetWidth)
+{
+    // The input parameters of this method rely on Measure::computeWidth(), so always call this method from there
+    const double epsilon = 0.1 * spatium();
+    static constexpr double multiplier = 1.4; // Empirical value for fastest convergence of the following loop
+    static constexpr int maxIter = 200; // Just for safety, in reality just 2 iterations are typically needed.
```

Comment:
Could you please explain why you use this particular number here? It is much greater than the number mentioned in the comment. It gives the feeling that this algorithm will work unpredictably

Reply:
The same question for 

static constexpr int maxIter = 400;

in LayoutSystem::collectSystem

Reply:
In any case, it is unclear why you used two different values for maxIter and why they are so big. Especially, considering this observation:

>I haven't seen so far any system taking more than 30-40 iterations (and only in very rare cases)
 
So, they should both be reduced to 40 (or even less, if this case is unrealistic). Ideally, we should use the same constant in both places

Reply:
400 and 200 are too big numbers. We can have two different constants, but their values should definitely be reduced to something more realistic

Reply:
I agree. Maybe 64. Would cover all known cases and still have enough headroom.

Reply:
Or 42, the answer to all questions ;-)

Reply:
Ok, I've gone back and tried to test as many edge cases as I could think of. For system justification I've seen a maximum of 50 iterations needed, for measure justification I've seen a maximum of 20. So `maxIter` is now set to 100 for systems, and 50 for measures. It really shouldn't be less than that: you _never_ design a system to fail on cases that may be just slightly harder than the hardest ones you tested. Also because `maxIter` is almost irrelevant in practice: if a system converges to the target after just 3 iterations, it will exit the loop at 3 iterations, regardless of `maxIter` being 100 or 1000.